src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -357,6 +358,93 @@ private static byte[] UrlEncode(byte[] bytes, int offset, int count)
             return expandedBytes;
         }
 
+        private static unsafe string UrlEncode(string originalString, char* s, int length)
```

Comment:
Nit: Maybe the name of the `char*` parameter should be changed to something like `pch`? I keep mistaking it for a string since it's named `s`.